unattended.sh

```diff
@@ -1,186 +1,789 @@
 #!/bin/sh
 
-function error_and_exit {
-  echo -e "$1"
-  exit 1
-}
+### Define all supported constants
+OPTIONS=":t:v:r:l:"
+declare -A SUPPORTED_LOG_LEVEL=([DEBUG]=0 [INFO]=1 [WARN]=2 [ERROR]=3)
+declare -A SUPPORTED_TOPOLOGY=([central]=1 [poller]=1)
+declare -A SUPPORTED_VERSION=([21.04]=1)
+declare -A SUPPORTED_REPOSITORY=([testing]=1 [unstable]=1 [stable]=1)
+default_timeout_in_sec=5
+script_short_name="$(basename $0)"
+default_ip=$(hostname -I | awk '{print $1}')
+###
+
+#Define default values
+
+passwords_file=/etc/centreon/generated.tobesecured                                        #File where the generated passwords will be temporaly saved
+tmp_passwords_file=/tmp/$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1) #Random tmp file as the /etc/centreon does not exist yet
```

Comment:
```suggestion
tmp_passwords_file=$(mktemp /tmp/passwd.XXXXXX) #Random tmp file as the /etc/centreon does not exist yet
```
Use the `mktemp` command which is exactly for this purpose